catroid/src/main/java/org/catrobat/catroid/content/actions/WriteListOnDeviceAction.kt

```diff
@@ -0,0 +1,64 @@
+/*
+ * Catroid: An on-device visual programming system for Android devices
+ * Copyright (C) 2010-2021 The Catrobat Team
+ * (<http://developer.catrobat.org/credits>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * An additional term exception under section 7 of the GNU Affero
+ * General Public License, version 3, is available at
+ * http://developer.catrobat.org/license_additional_term
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.catrobat.catroid.content.actions
+
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+import org.catrobat.catroid.ProjectManager
+import org.catrobat.catroid.formulaeditor.UserList
+import org.catrobat.catroid.io.DeviceListAccessor
+import org.catrobat.catroid.io.DeviceUserDataAccessor
+
+class WriteListOnDeviceAction : AsynchronousAction() {
```

Comment:
Since WriteListOnDeviceAction and WriteVariableOnDeviceAction are basically the same class, you could dependency inject the user-list/variable (UserData) and the Accessor to a new class WriteUserDataOnDeviceAction. Also, please inject the coroutine scope.

Reply:
https://developer.android.com/training/dependency-injection
(Please, don't use the automated injection)

Reply:
https://developer.android.com/training/dependency-injection
(Please, don't use the automated injection)